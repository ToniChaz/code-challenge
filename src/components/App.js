import React, { Component } from 'react'
import { ARTICLES_QUERY } from '../constants/querys/queries'

class App extends Component {
  componentWillMount() {
    this.props.fetchArticles(ARTICLES_QUERY)
  }

  render() {
    const { articles } = this.props

    if (!articles) return null

    return (
      <div className="App">
        <h2>Billin code challenge</h2>
        <pre>{JSON.stringify(articles, null, 2)}</pre>
      </div>
    )
  }
}

export default App
