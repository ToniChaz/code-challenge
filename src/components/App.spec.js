// App.spec.js

import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import App from './App'

describe('Testing App', () => {
  test('Renders correctly', () => {
    const tree = shallow(
      <App />
    )
    expect(shallowToJson(tree)).toMatchSnapshot()
  })
})
