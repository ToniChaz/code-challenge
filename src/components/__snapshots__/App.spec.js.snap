exports[`Testing App Renders correctly 1`] = `
<Container
  tag="div">
  <Connect(Header) />
  <Connect(Articles) />
  <Connect(Footer) />
</Container>
`;
